lib/puppet_litmus/serverspec.rb

```diff
@@ -146,17 +152,23 @@ def run_bolt_task(task_name, params = {}, opts = {})
 
     result = run_task(task_name, target_node_name, params, config: config_data, inventory: inventory_hash)
 
-    raise "task failed\n`#{task_name}`\n======\n#{result}" if result.first['status'] != 'success' && opts[:expect_failures] != true
+    result_obj = {
+      exit_code: 0,
+      stdout: (result.first['result']['_output'] unless result.first['result']['_output'].nil?),
+      stderr: nil,
+      result: result.first['result'],
+    }
+
+    if result.first['status'] != 'success'
+      raise "task failed\n`#{task_name}`\n======\n#{result}" if opts[:expect_failures] != true
+      result_obj[:exit_code] = result.first['result']['_error']['details']['exitcode'].nil? ? 255 : result.first['result']['_error']['details']['exitcode']
+      result_obj[:stderr]    = result.first['result']['_error']['msg']
+    end
 
-    exit_code = if result.first['status'] == 'success'
-                  0
-                else
-                  255
-                end
-    result = OpenStruct.new(exit_code: exit_code,
-                            stdout: result.first['result']['status'],
-                            stderr: result.first['result']['stderr'],
-                            result: result.first['result'])
+    result = OpenStruct.new(exit_code: result_obj[:exit_code],
```

Comment:
Not super important but I was wondering why we do this instead of just returning a hash?

Reply:
because beaker :(, any tests written rely on a result object. and it would have meant large scale changes across module tests.

Reply:
![](https://media.giphy.com/media/k7V9BtRBXEby25fxa8/giphy.gif)